test/optimizer/table_or_pushdown.test

```diff
@@ -0,0 +1,96 @@
+# name: test/optimizer/table_or_pushdown.test
+# description: Test Table OR Filter Push Down
+# group: [optimizer]
+
+statement ok
+CREATE TABLE integers AS SELECT a as A, a as B FROM generate_series(1, 5, 1) tbl(a)
+
+# test OR filters with multiple columns in the root OR, e.g., a=1 OR b=2
+query II
+EXPLAIN SELECT * FROM integers WHERE a=1 OR b=2 AND (a>3 OR b<5)
+----
+physical_plan	<REGEX>:.*SEQ_SCAN.*integers.*a[ ]*|.*b[ ]*|.*
+
+query II
+SELECT * FROM integers WHERE a=1 OR b=2 AND (a>3 OR b<5)
+----
+1	1
+2	2
+
+
+# test OR filters with AND that triggers a stop early condition
+query II
+EXPLAIN SELECT * FROM integers WHERE a=1 OR a=2 AND (a>3 OR b<5)
+----
+physical_plan	<REGEX>:.*SEQ_SCAN.*Filters: a=1 OR a=2[ ]*|.*
+
+query II
+SELECT * FROM integers WHERE a=1 OR a=2 AND (a>3 OR b<5)
+----
+1	1
+2	2
+
+
+# test OR filters with AND in the same column
+query II
+EXPLAIN SELECT * FROM integers WHERE a=1 OR (a>3 AND a<5)
+----
+physical_plan	<REGEX>:.*SEQ_SCAN.*Filters: a=1 OR a>3 AND a<5|.*
+
+query II
+SELECT * FROM integers WHERE a=1 OR (a>3 AND a<5)
+----
+1	1
+4	4
+
+# test only OR filters
+query II
+EXPLAIN SELECT * FROM integers WHERE a=1 OR a>3 OR a<5
+----
+physical_plan	<REGEX>:.*SEQ_SCAN.*Filters: a=1 OR a>3 OR a<5|.*
+
+query II
+SELECT * FROM integers WHERE a=1 OR a>3 OR a<5 ORDER by a
```

Comment:
Could we add more tests with more complex `AND/OR` chains? And tests with more complex functions/conditions (e.g. `a=1 IS NULL`, `concat(a=1, b=1)='11'`, etc) within OR/AND. Case expressions (`CASE WHEN x=7 THEN ...`) within conjunctions.